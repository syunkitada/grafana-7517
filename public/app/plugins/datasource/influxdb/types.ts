import { DataQuery, DataSourceJsonData } from '@grafana/data';

export interface InfluxOptions extends DataSourceJsonData {
  timeInterval: string;
  httpMode: string;

  // Influx 2.0
  enableFlux?: boolean;
  organization?: string;
  defaultBucket?: string;
  maxSeries?: number;
}

export interface InfluxSecureJsonData {
  password?: string;
  token?: string;
}

export interface InfluxQueryPart {
  type: string;
  params?: string[];
  interval?: string;
}

export interface InfluxQueryTag {
  key: string;
  operator?: string;
  condition?: string;
  value: string;
}

export enum InfluxQueryType {
  Classic = 'Classic', // IFQL query builder
  InfluxQL = 'InfluxQL', // raw ifql
  Flux = 'Flux',
}

export interface InfluxQuery extends DataQuery {
  queryType?: InfluxQueryType;
  policy?: string;
  measurement?: string;
  resultFormat?: 'time_series' | 'table';
  orderByTime?: string;
  tags?: InfluxQueryTag[];
  groupBy?: InfluxQueryPart[];
  select?: InfluxQueryPart[][];
  limit?: string;
  slimit?: string;
  tz?: string;
  fill?: string;
  rawQuery?: boolean; // deprecated (use raw InfluxQL)
  query?: string;
}
