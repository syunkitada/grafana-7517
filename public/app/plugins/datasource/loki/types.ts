import { DataQuery } from '@grafana/ui/src/types';

export interface LokiQuery extends DataQuery {
  expr: string;
}

