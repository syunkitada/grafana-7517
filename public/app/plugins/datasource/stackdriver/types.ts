export enum MetricFindQueryTypes {
  Services = 'services',
  MetricTypes = 'metricTypes',
  MetricLabels = 'metricLabels',
  ResourceLabels = 'resourceLabels',
  ResourceTypes = 'resourceTypes',
  Aggregations = 'aggregations',
  Alignerns = 'alignerns',
  AlignmentPeriods = 'alignmentPeriods',
}

export interface TemplateQueryComponentData {
  selectedQueryType: string;
  metricDescriptors: any[];
  selectedService: string;
  selectedMetricType: string;
  labels: string[];
  labelKey: string;
  metricTypes: Array<{ value: string; name: string }>;
  services: Array<{ value: string; name: string }>;
}
