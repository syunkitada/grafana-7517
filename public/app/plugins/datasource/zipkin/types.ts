export type ZipkinSpan = {
  traceId: string;
  parentId?: string;
  name: string;
  id: string;
  timestamp: number;
  duration: number;
  localEndpoint: {
    serviceName: string;
    ipv4: string;
    port?: number;
  };
  annotations?: ZipkinAnnotation[];
  tags?: { [key: string]: string };
  kind?: 'CLIENT' | 'SERVER' | 'PRODUCER' | 'CONSUMER';
};

export type ZipkinAnnotation = {
  timestamp: number;
  value: string;
};
