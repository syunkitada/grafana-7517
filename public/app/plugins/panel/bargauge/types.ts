import { SingleStatBaseOptions, BarGaugeDisplayMode } from '@grafana/ui';
import { commonValueOptionDefaults } from '../stat/types';
import { VizOrientation, SelectableValue } from '@grafana/data';

export interface BarGaugeOptions extends SingleStatBaseOptions {
  displayMode: BarGaugeDisplayMode;
  showUnfilled: boolean;
}

export const displayModes: Array<SelectableValue<string>> = [
  { value: BarGaugeDisplayMode.Gradient, label: 'Gradient' },
  { value: BarGaugeDisplayMode.Lcd, label: 'Retro LCD' },
  { value: BarGaugeDisplayMode.Basic, label: 'Basic' },
];

export const defaults: BarGaugeOptions = {
  displayMode: BarGaugeDisplayMode.Lcd,
  orientation: VizOrientation.Horizontal,
  reduceOptions: commonValueOptionDefaults,
  showUnfilled: true,
};
