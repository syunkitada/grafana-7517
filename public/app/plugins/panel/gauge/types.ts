import { VizOrientation, SelectableValue } from '@grafana/data';
import { SingleStatBaseOptions } from '@grafana/ui/src/components/SingleStatShared/SingleStatBaseOptions';
import { commonValueOptionDefaults } from '../stat/types';

export interface GaugeOptions extends SingleStatBaseOptions {
  showThresholdLabels: boolean;
  showThresholdMarkers: boolean;
}

export const orientationOptions: Array<SelectableValue<VizOrientation>> = [
  { value: VizOrientation.Auto, label: 'Auto' },
  { value: VizOrientation.Horizontal, label: 'Horizontal' },
  { value: VizOrientation.Vertical, label: 'Vertical' },
];

export const defaults: GaugeOptions = {
  showThresholdMarkers: true,
  showThresholdLabels: false,
  reduceOptions: commonValueOptionDefaults,
  orientation: VizOrientation.Auto,
};
