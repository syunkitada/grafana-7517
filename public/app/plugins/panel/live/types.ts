import { LiveChannelAddress } from '@grafana/data';

export interface LivePanelOptions {
  channel?: LiveChannelAddress;
}
