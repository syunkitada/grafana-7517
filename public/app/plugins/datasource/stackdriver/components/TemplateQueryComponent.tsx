import React, { PureComponent } from 'react';
import StackdriverDatasource from '../datasource';
import ServiceSelector from './ServiceSelector';
import MetricTypeSelector from './MetricTypeSelector';

interface Props {
  datasource: StackdriverDatasource;
  query: any;
  onChange: (c: string) => void;
}

export class StackdriverTemplateQueryComponent extends PureComponent<Props, any> {
  queryTypes: Array<{ value: string; name: string }> = [
    { value: 'services', name: 'Services' },
    { value: 'metricTypes', name: 'Metric Types' },
    { value: 'metricLabels', name: 'Metric labels For Metric Type' },
  ];

  constructor(props) {
    super(props);
    this.handleQueryTypeChange = this.handleQueryTypeChange.bind(this);
    this.onServiceChange = this.onServiceChange.bind(this);
    this.onMetricTypeChange = this.onMetricTypeChange.bind(this);
    this.state = { queryType: undefined, metricDescriptors: [], service: undefined, metricType: undefined };
  }

  async componentDidMount() {
    const metricDescriptors = await this.props.datasource.getMetricTypes(this.props.datasource.projectName);
    this.setState({ metricDescriptors });
  }

  handleQueryTypeChange(event) {
    this.setState({ queryType: event.target.value });
  }

  onServiceChange(event) {
    this.setState({ service: event.target.value });
  }

  onMetricTypeChange(event) {
    this.setState({ metricType: event.target.value });
  }

  renderSwitch(queryType) {
    switch (queryType) {
      case 'metricTypes':
        return (
          <ServiceSelector metricDescriptors={this.state.metricDescriptors} onServiceChange={this.onServiceChange} />
        );
      case 'metricLabels':
        return (
          <React.Fragment>
            <ServiceSelector metricDescriptors={this.state.metricDescriptors} onServiceChange={this.onServiceChange} />
            <MetricTypeSelector
              selectedService={this.state.service}
              metricDescriptors={this.state.metricDescriptors}
              onMetricTypeChange={this.onMetricTypeChange}
            />
          </React.Fragment>
        );
      default:
        return '';
    }
  }

  render() {
    return (
      <React.Fragment>
        <div className="gf-form max-width-21">
          <span className="gf-form-label width-7">Query Type</span>
          <div className="gf-form-select-wrapper max-width-14">
            <select className="gf-form-input" required onChange={this.handleQueryTypeChange}>
              {this.queryTypes.map((qt, i) => (
                <option key={i} value={qt.value} ng-if="false">
                  {qt.name}
                </option>
              ))}
            </select>
          </div>
        </div>
        {this.renderSwitch(this.state.queryType)}
      </React.Fragment>
    );
  }
}
