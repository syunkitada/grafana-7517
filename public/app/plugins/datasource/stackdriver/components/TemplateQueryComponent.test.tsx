import React from 'react';
import renderer from 'react-test-renderer';
import { StackdriverTemplateQueryComponent } from './TemplateQueryComponent';
import { TemplateQueryProps } from 'app/types/plugins';

jest.mock('../functions', () => ({
  getMetricTypes: () => Promise.resolve({ metricTypes: [], selectedMetricType: '' }),
  extractServicesFromMetricDescriptors: m => m,
}));

const props: TemplateQueryProps = {
  onChange: (query, definition) => {},
  query: '',
  datasource: {
    getMetricTypes: async p => [],
  },
};

describe('StackdriverTemplateQueryComponent', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<StackdriverTemplateQueryComponent {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should use the first query type in the array if no query type was saved before', done => {
    props.onChange = (query, definition) => {
      expect(definition).toBe('Stackdriver - Services');
      done();
    };
    renderer.create(<StackdriverTemplateQueryComponent {...props} />).toJSON();
  });
});
