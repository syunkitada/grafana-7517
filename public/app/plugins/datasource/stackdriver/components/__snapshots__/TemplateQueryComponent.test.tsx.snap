// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackdriverTemplateQueryComponent renders correctly 1`] = `
Array [
  <div
    className="gf-form max-width-21"
  >
    <span
      className="gf-form-label width-10"
    >
      Query Types
    </span>
    <div
      className="gf-form-select-wrapper max-width-10"
    >
      <select
        className="gf-form-input"
        onChange={[Function]}
        required={true}
        value="metricTypes"
      >
        <option
          value="metricTypes"
        >
          Metric Types
        </option>
        <option
          value="metricLabels"
        >
          Metric Labels
        </option>
        <option
          value="resourceLabels"
        >
          Resource Labels
        </option>
        <option
          value="resourceTypes"
        >
          Resource Types
        </option>
        <option
          value="aggregations"
        >
          Aggregations
        </option>
        <option
          value="aligners"
        >
          Aligners
        </option>
        <option
          value="alignmentPeriods"
        >
          Alignment Periods
        </option>
      </select>
    </div>
  </div>,
  <div
    className="gf-form max-width-21"
  >
    <span
      className="gf-form-label width-10"
    >
      Services
    </span>
    <div
      className="gf-form-select-wrapper max-width-10"
    >
      <select
        className="gf-form-input"
        onChange={[Function]}
        required={true}
        value=""
      />
    </div>
  </div>,
]
`;
